R/conditions.R

```diff
@@ -365,21 +365,64 @@ mismatchError <- function(name_specified, name_null,
 
 ###Added by ME 5/9/22
 
+# TODO tests
 #' Datatable does not exist error.
 #'
 #' @param dtName Names of the dt.
 #' @param call sys.call to pass on to the error.
 #' @param msg Additional information for the error message as
 #' glueCollapse'able string.
 #' @noRd
-dtDoesNotExistError <- function(dtNameStr, call = sys.call(-1), msg = "", ...) {
+# dtDoesNotExistError <- function(dtNameStr, call = sys.call(-1), msg = "", ...) {
+#   message <- glueCollapse(
+#     "Datatable { dtNameStr *} not defined!", msg
+#   )
+#   
+#   c <- condition(
+#     c("simstudy::dtDoesNotExist", "error"),
+#     message, call, ...
+#   )
+#   stop(c)
+# }
+
+###Added by ME 5/12/22
+
+# TODO tests
+#' Datatable does not exist error.
+#'
+#' @param dtName Names of the dt.
+#' @param call sys.call to pass on to the error.
+#' @param msg Additional information for the error message as
+#' glueCollapse'able string.
+#' @noRd
+# coeffVarError <- function(call = sys.call(-1), msg = "", ...) {
+#   message <- glueCollapse(
+#     "Coefficients or variables not properly specified!", msg
+#   )
+#   
+#   c <- condition(
+#     c("simstudy::coeffVar", "error"),
+#     message, call, ...
+#   )
+#   stop(c)
+# }
+
+#' Variables not equal to value error.
+#'
+#' @param vars Names of the variables.
+#' @param val Value to be equal to.
+#' @param call sys.call to pass on to the error.
+#' @param msg Additional information for the error message as
+#' glueCollapse'able string.
+#' @noRd
+notEqualError <- function(vars, val, call = sys.call(-1), msg = "", ...) {
   message <- glueCollapse(
-    "Datatable { dtNameStr *} not defined!", msg
+    "{ vars *} not equal to {val}!", msg
   )
-  
+
   c <- condition(
-    c("simstudy::dtDoesNotExist", "error"),
+    c("simstudy::notEqual", "error"),
     message, call, ...
   )
   stop(c)
-}
\ No newline at end of file
+}
```

Comment:
The argument `vars` in `notEqualError was causing me problems, because you want the names, so I've modified the code:

```
notEqualError <- function(names, val, call = sys.call(-1), msg = "", ...) {

   message <- glueCollapse(
         "{ names *} not equal to {val}!", msg
   )

   c <- condition(
      c("simstudy::notEqual", "error"),
       message, call, ...
     )

   stop(c)
}
```

We should really remove the `msg` argument from here, or at least make it work better with the calling function.